lib/common/compiler.h

```diff
@@ -92,7 +92,7 @@
 #if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_I86))  /* _mm_prefetch() is not defined outside of x86/x64 */
 #  include <mmintrin.h>   /* https://msdn.microsoft.com/fr-fr/library/84szxsww(v=vs.90).aspx */
 #  define PREFETCH(ptr)   _mm_prefetch((const char*)ptr, _MM_HINT_T0)
-#elif defined(__GNUC__)
+#elif defined(__GNUC__) && ((__GNUC__ >= 4) || (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1))
```

Comment:
add parenthesis for `|| ( (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1) )`

Reply:
Done.
BTW, shall I squash my 2 commits?

Reply:
As you wish.
It's a bit better if they are squashed,
but not an absolute necessity.